backend/app/services/data_provider_configurations.rb

```diff
@@ -20,13 +20,16 @@ def config_for(key)
     end
   end
 
-  private
-
   def config
-    @config ||= YAML.load_file(config_file, aliases: true)
+    @config ||= YAML.safe_load(config_raw, aliases: true)
   end
 
-  def config_file
-    Rails.root.join("config/data_providers.yml")
+  def config_raw
+    @file_names = Dir.glob("config/data_providers/*")
+    @yml_content = ""
+    @file_names.each do |file_name|
+      @yml_content += File.read(file_name)
+    end
+    @yml_content
   end
```

Comment:
```suggestion
  def config_raw
    data_providers_files.inject('') do |string, endpoint_file|
      string + File.read(endpoint_file)
    end
  end
  
  def data_providers_files
    Dir.glob("config/data_providers/*")
  end
```

1. Les variables d'instances ne servent pas
2. Faire 2 méthodes me semble plus propre et lisible
3. Un inject me semble plus fonctionnel